File: src/test/java/json4dummies/JsonTest.java
package json4dummies;

import com.google.common.collect.ImmutableMap;
import org.junit.Test;

import java.util.Map;

import static json4dummies.Json.fromJson;
import static json4dummies.Json.toJson;
import static json4dummies.Json.toJsonBytes;
import static org.assertj.core.api.Assertions.assertThat;

public class JsonTest {

    Map map = ImmutableMap.of("foo", "bar");

    @Test
    public void shouldConvertMap() {
        String json = toJson(map);
        Map<String, Object> parsedMap = fromJson(json);
        assertThat(parsedMap).isEqualTo(map);
    }

    @Test
    public void shouldConvertMapToBytes() {
        byte[] json = toJsonBytes(map);
        Map<String, Object> parsedMap = fromJson(json);
        assertThat(parsedMap).isEqualTo(map);
    }

    @Test
    public void shouldReadBytes() {
        String json = toJson(map);
        Map<String, Object> parsedMap = fromJson(json.getBytes());
        assertThat(parsedMap).isEqualTo(map);
    }

    @Test
    public void shouldIncludeNullFields() {
        String json = toJson(new Person(null, null));
        assertThat(json).contains("\"name\":null");
        assertThat(json).contains("\"surname\":null");
    }

}

File: src/test/kotlin/json4dummies/JsonTest.kt
package json4dummies;

import json4dummies.Json.Companion.fromJson
import json4dummies.Json.Companion.toJson
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class JsonTest {

    val map = mapOf("foo" to "bar")

    @Test
    fun shouldConvertMapToJson() {
        // When
        val json = toJson(map)

        // Then
        val parsedMap = fromJson(json)
        assertThat(parsedMap).isEqualTo(map)
    }

    @Test
    fun shouldConvertMapToBytesToJson() {
        val json = Json.toJsonBytes(map)
        val parsedMap = Json.fromJson(json)
        assertThat(parsedMap).isEqualTo(map)
    }

    @Test
    fun shouldReadBytes() {
        val json = toJson(map)
        val parsedMap = fromJson(json.toByteArray())
        assertThat(parsedMap).isEqualTo(map);
    }

    @Test
    fun shouldIncludeNullFields() {
        val json = toJson(Person(null, null))
        assertThat(json).contains("\"name\":null")
        assertThat(json).contains("\"surname\":null")
    }

}
